moz_security/sandboxes/heka/analysis/moz_security_hh_cms.lua

```diff
@@ -0,0 +1,154 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+#  Mozilla Security Heavy Hitters, CMS + moving average
+
+For events matching the message_matcher, this analysis plugin attempts to calculate typical
+request rates while identifying and flagging anomolous request patterns.
+
+Request counts within a given window are stored in a Count-Min sketch data structure; based on
+a sample encountered during the window this data structure is used to consult request rates for
+given identifiers, where an identifier exceeds a threshold it is added to an analysis list for
+submission by the plugin.
+
+## Sample Configuration
+```lua
+filename = "moz_security_hh_cms.lua"
+message_matcher = "Logger == 'input.nginx'"
+ticker_interval = 60
+preserve_data = false -- This module cannot keep state at this time
+
+id_field = "Fields[remote_addr]" -- field to use as the identifier
+-- id_field_capture = ",? *([^,]+)$",  -- optional e.g. extract the last entry in a comma delimited list
+
+sample_min_id = 5000 -- Minimum distinct identifers before a sample will be calculated
+sample_max_id = 10000 -- Maximum identifiers to use for sample calculation within a window
+sample_min_ev = 50000 -- Minimum number of events sampler must consume before calculation
+sample_window = 60 -- Sample window size
+sample_ticks = 2500 -- Recalculate sample every sample_ticks events
+threshold_cap = 10 -- Threshold will be calculated average + (calculated average * cap)
+-- cms_epsilon = 1 / 10000 -- optional CMS value for epsilon
+-- cms_delta = 0.0001 -- optional CMS value for delta
+```
+--]]
+
+require "string"
+require "table"
+
+local c         = require "streaming_algorithms.cm_sketch"
```

Comment:
This is only used once so the non-descriptive c. mapping can be removed and simply call
```lua
local cms = streaming_algorithms.cm_sketch.new(cms_epsilon, cms_delta)
```
below